Scripts/PAIF-N/01-deploy-vcf-workload-domain.ps1

```diff
@@ -0,0 +1,345 @@
+      Vip = '10.0.0.44'
+      VipFqdn = "$domainName-nsx01.$domain"
+   }
+}
+# --------------------------------------------------------------------------------------------------------------------------
+
+## Workload domain spec construction
+$esxiFqdns = $esxiHosts | Select-Object -ExpandProperty Fqdn
+$VcfHost = Invoke-VcfGetHosts -status "UNASSIGNED_USEABLE" | `
+   Select-Object -ExpandProperty Elements | `
+   Where-Object { $esxiFqdns -contains $_.Fqdn }
+
+$ComputeSpec = Initialize-VcfComputeSpec -ClusterSpecs (
+   $domainSpec.ComputeSpec.ClusterSpecs | ForEach-Object {
+      $_clusterSpec = $_
+
+      Initialize-VcfClusterSpec `
+         -DatastoreSpec (
+            Initialize-VcfDatastoreSpec `
+               -VsanDatastoreSpec (
+                  Initialize-VcfVsanDatastoreSpec `
+                     -DatastoreName $_.DatastoreSpec.VSanDatastoreSpec.DatastoreName `
+                     -FailuresToTolerate $_.DatastoreSpec.VSanDatastoreSpec.FailuresToTolerate `
+                     -LicenseKey $_.DatastoreSpec.VSanDatastoreSpec.LicenseKey
+               )) `
+         -HostSpecs (
+            $VcfHost | ForEach-Object {
+               $esxiHost = $esxiHosts | Where-Object { $_.Fqdn -eq $_.Fqdn } | Select-Object -First 1
+
+               Initialize-VcfHostSpec `
+                  -Id $_.Id `
+                  -LicenseKey $esxiHost.LicenseKey `
+                  -HostNetworkSpec (
+                     Initialize-VcfHostNetworkSpec -VmNics @(
+                        Initialize-VcfVmNic -Id 'vmnic0' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[0].Name
+                        Initialize-VcfVmNic -Id 'vmnic1' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[0].Name
+                        Initialize-VcfVmNic -Id 'vmnic2' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[1].Name
+                        Initialize-VcfVmNic -Id 'vmnic3' -VdsName $_clusterSpec.NetworkSpec.VdsSpecs[1].Name
+                     ))
+            }
+         ) `
+         -Name $_.Name `
+         -NetworkSpec (
+            Initialize-VcfNetworkSpec `
+               -NsxClusterSpec (
+                  Initialize-VcfNsxClusterSpec -NsxTClusterSpec (
+                     Initialize-VcfNsxTClusterSpec `
+                        -GeneveVlanId $_.NetworkSpec.NsxClusterSpec.NsxTClusterSpec.GeneveVlanId
+                        )
+                  ) `
+               -VdsSpecs @(
+                  Initialize-VcfVdsSpec -Name $_.NetworkSpec.VdsSpecs[0].Name `
+                     -PortGroupSpecs (
+                        $_.NetworkSpec.VdsSpecs[0].PortGroups | ForEach-Object {
+                           Initialize-VcfPortgroupSpec `
+                              -Name $_.Name `
+                              -TransportType $_.TransportType
+                        })
+                  Initialize-VcfVdsSpec -Name $_.NetworkSpec.VdsSpecs[1].Name `
+                     -IsUsedByNsxt $_.NetworkSpec.VdsSpecs[1].IsUsedByNsxt
```

Comment:
Why not have the port groups from the spec be created for this second switch?

Reply:
It is used for workloads only.